x/csr/keeper/begin_block_test.go

```diff
@@ -0,0 +1,26 @@
+package keeper_test
+
+import (
+	"github.com/Canto-Network/Canto/v2/x/csr/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Test Genesis tests that the Turnstile has been deployed on genesis, and
+// that the module account exists
+func (suite *KeeperTestSuite) TestGenesis() {
+	// turnstile should not exist before begin block
+	_, found := suite.app.CSRKeeper.GetTurnstile(suite.ctx)
+	suite.Require().False(found)
+	// begin new block
+	suite.Commit()
+	// Get the Turnstile address, and check that there is indeed code at the address
+	turnstile, found := suite.app.CSRKeeper.GetTurnstile(suite.ctx)
+	suite.Require().True(found) 
+	acc := suite.app.EvmKeeper.GetAccountWithoutBalance(suite.ctx, turnstile)
+	// check that there is indeed code at this address
+	suite.Require().NotEqual(acc.CodeHash, crypto.Keccak256(nil))
```

Comment:
can add the same check to see if the smart contract bin matches the turnstile match